File: model/document.py
"""Document ORM model."""
from __future__ import annotations
from typing import Optional

from sqlalchemy import ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from . import Base


class Document(Base):
    __tablename__ = "documents"

    document_id: Mapped[str] = mapped_column(
        String(64),
        ForeignKey("consignments.consignment_num", ondelete="CASCADE"),
        primary_key=True,
        nullable=False,
    )
    pod: Mapped[Optional[str]] = mapped_column(Text)
    invoice: Mapped[Optional[str]] = mapped_column(Text)

    consignment: Mapped["Consignment"] = relationship(
        "Consignment",
        back_populates="document",
        passive_deletes=True,
    )

File: model/document.js
export class Document {
  static tableName = 'documents';
  static schema = {
    documentId: { column: 'document_id', type: 'string', primaryKey: true, required: true, maxLength: 64, references: 'consignments.consignment_num', onDelete: 'CASCADE' },
    pod: { type: 'text' },
    invoice: { type: 'text' },
  };

  constructor(attrs = {}) {
    Object.assign(this, attrs);
    this.documentId = attrs.documentId ?? attrs.document_id;
  }
}
